storagemarket/logs/log.go

```diff
@@ -15,54 +16,70 @@ type DealLogger struct {
 	ctx    context.Context
 	logger *logging.ZapEventLogger
 	logsDB *db.LogsDB
+	prefix string
 }
 
 func NewDealLogger(ctx context.Context, logsDB *db.LogsDB) *DealLogger {
+	prefix := "boost-storage-deal"
 	return &DealLogger{
 		ctx:    ctx,
-		logger: logging.Logger("boost-storage-deal"),
+		logger: logging.Logger(prefix),
 		logsDB: logsDB,
+		prefix: prefix,
 	}
 }
 
-func (d *DealLogger) Infow(dealId uuid.UUID, msg string, kvs ...interface{}) {
-	d.logger.Infow(msg, "id", dealId, kvs)
-	l := &db.DealLog{
-		DealUUID:  dealId,
-		CreatedAt: time.Now(),
-		LogLevel:  "INFO",
-		LogMsg:    msg,
-		LogParams: fmt.Sprintf("%+v", kvs),
+func (d *DealLogger) Subsystem(name string) *DealLogger {
+	prefix := d.prefix + "/" + name
+	return &DealLogger{
+		ctx:    d.ctx,
+		logger: logging.Logger(prefix),
+		logsDB: d.logsDB,
+		prefix: prefix,
 	}
+}
 
-	if err := d.logsDB.InsertLog(d.ctx, l); err != nil {
-		d.logger.Warnw("failed to persist deal log", "id", dealId, "err", err)
-	}
+func (d *DealLogger) Infow(dealId uuid.UUID, msg string, kvs ...interface{}) {
+	d.logger.Infow(msg, "id", dealId, kvs)
+	d.updateLogDB(dealId, msg, "INFO", kvs)
 }
 
 func (d *DealLogger) Warnw(dealId uuid.UUID, msg string, kvs ...interface{}) {
 	d.logger.Warnw(msg, "id", dealId, kvs)
+	d.updateLogDB(dealId, msg, "WARN", kvs)
+}
+
+func (d *DealLogger) Errorw(dealId uuid.UUID, errMsg string, kvs ...interface{}) {
+	d.logger.Errorw(errMsg, "id", dealId, kvs)
+	d.updateLogDB(dealId, errMsg, "ERROR", kvs)
+}
+
+func (d *DealLogger) updateLogDB(dealId uuid.UUID, msg string, level string, kvs ...interface{}) {
+	jsn, err := json.Marshal(kvs)
+	if err != nil {
+		d.logger.Warnw("failed to marshal log params to json", "err", err, "id", dealId)
+	}
 
 	l := &db.DealLog{
 		DealUUID:  dealId,
 		CreatedAt: time.Now(),
-		LogLevel:  "WARN",
-		LogMsg:    msg,
-		LogParams: fmt.Sprintf("%+v", kvs),
+		LogLevel:  level,
+		LogMsg:    d.prefix + ":" + msg,
```

Comment:
I'm not sure we want the prefix here unless there's a subsytem. Otherwise all deal logs are going to start with the same prefix, which seems redundant.
Maybe we should create a separate DB field called "subsystem" that's empty for top level deal logs, and has the subsystem for next level deal logs. eg
```
dealLog.Infow("msg") => Subsystem: "", LogMsg: "msg"
dealLog.Subsystem("sub").Infow("msg") => Subsystem: "sub", LogMsg: "msg"
```